homeassistant/helpers/template.py

```diff
@@ -329,10 +330,22 @@ def async_render(self, variables: TemplateVarsType = None, **kwargs: Any) -> str
             kwargs.update(variables)
 
         try:
-            return compiled.render(kwargs).strip()
+            render_result = compiled.render(kwargs)
         except jinja2.TemplateError as err:
             raise TemplateError(err) from err
 
+        if not self.hass.config.legacy_templates:
+            try:
+                result = literal_eval(render_result)
```

Comment:
We should strip this or we get "invalid indent" errors if some white space ends up in front. 

Any reason we wouldn't just always call strip on `render_result`?

Reply:
Adjusted in 425b2ad, now always stripping before use.